mne/viz/_figure.py

```diff
@@ -0,0 +1,1854 @@
+            ('-', 'Decrease signal scaling'),
+            ('b', 'Toggle butterfly mode' if inst != 'ica' else None),
+            ('d', 'Toggle DC removal' if is_raw else None),
+            ('w', noise_cov),
+            ('h', 'Show peak-to-peak histogram' if inst == 'epochs' else None),
+            ('_USER INTERFACE', ' '),
+            ('a', 'Toggle annotation mode' if is_raw else None),
+            ('j', 'Toggle SSP projector window' if is_raw else None),
+            ('p', 'Toggle draggable annotations' if is_raw else None),
+            ('s', 'Toggle scalebars' if inst != 'ica' else None),
+            ('z', 'Toggle scrollbars'),
+            ('F11', 'Toggle fullscreen'),
+            ('?', 'Open this help window'),
+            ('esc', 'Close focused figure or dialog window'),
+            ('_MOUSE INTERACTION', ' '),
+            (f'Left-click {ch_cmp} name', lclick_name),
+            (f'Left-click {ch_cmp} data', lclick_data),
+            ('Left-click-and-drag on plot', ldrag),
+            ('Left-click on plot background', 'Place vertical guide'),
+            ('Right-click on plot background', 'Clear vertical guide'),
+            ('Right-click on channel name', None)  # TODO: rclick_name
+        ])
+        return help_text
+
+    # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+    # ANNOTATIONS
+    # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+
+    def _create_annotation_fig(self):
+        """Create the annotation dialog window."""
+        from matplotlib.widgets import Button, SpanSelector, CheckButtons
+        from mpl_toolkits.axes_grid1.axes_size import Fixed
+        from mpl_toolkits.axes_grid1.axes_divider import make_axes_locatable
+        # make figure
+        labels = np.array(sorted(set(self.mne.inst.annotations.description)))
+        width, var_height, fixed_height, pad = \
+            self._compute_annotation_figsize(len(labels))
+        figsize = (width, var_height + fixed_height)
+        fig = self._new_child_figure(figsize=figsize,
+                                     FigureClass=MNEAnnotationFigure,
+                                     fig_name='fig_annotation',
+                                     window_title='Annotations')
+        # make main axes
+        left = fig._inch_to_rel(pad)
+        bottom = fig._inch_to_rel(pad, horiz=False)
+        width = 1 - 2 * left
+        height = 1 - 2 * bottom
+        fig.mne.radio_ax = fig.add_axes((left, bottom, width, height),
+                                        frame_on=False, aspect='equal')
+        div = make_axes_locatable(fig.mne.radio_ax)
+        self._update_annotation_fig()  # populate w/ radio buttons & labels
+        # append instructions at top
+        instructions_ax = div.append_axes(position='top', size=Fixed(1),
+                                          pad=Fixed(5 * pad))
+        # XXX when we support a newer matplotlib (3.0?) the instructions can
```

Comment:
3.0.0 came out Sept 18 2018, and we're going to release next in March 2021. So it will have been out for 2.5 years then, which seems within our limit (I think we go with 2-ish years?). So if it makes things better we should consider upping our minimum, but maybe in another PR.

Reply:
See #8374 -- my vote would be to merge this PR then that one, to make it so you don't have to do another rebase here

Reply:
ok with me.  just waiting on @agramfort to sign off here.